hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFlusher.java

```diff
@@ -73,20 +73,20 @@ protected void finalizeWriter(StoreFileWriter writer, long cacheFlushSeqNum,
   /**
    * Creates the scanner for flushing snapshot. Also calls coprocessors.
    * @param snapshotScanners
-   * @param smallestReadPoint
    * @return The scanner; null if coprocessor is canceling the flush.
    */
   protected final InternalScanner createScanner(List<KeyValueScanner> snapshotScanners,
-      long smallestReadPoint, FlushLifeCycleTracker tracker) throws IOException {
+    FlushLifeCycleTracker tracker) throws IOException {
     ScanInfo scanInfo;
     if (store.getCoprocessorHost() != null) {
       scanInfo = store.getCoprocessorHost().preFlushScannerOpen(store, tracker);
     } else {
       scanInfo = store.getScanInfo();
     }
+    final long smallestReadPoint = store.getSmallestReadPoint();
     InternalScanner scanner = new StoreScanner(store, scanInfo, snapshotScanners,
         ScanType.COMPACT_RETAIN_DELETES, smallestReadPoint, HConstants.OLDEST_TIMESTAMP);
-    assert scanner != null;
```

Comment:
Any reason for doing this?

Reply:
As far as I can see, scanner can't be null here as it's a local variance, see below
```
scanner = new StoreScanner(store, scanInfo, snapshotScanners,
        ScanType.COMPACT_RETAIN_DELETES, smallestReadPoint, HConstants.OLDEST_TIMESTAMP);
```
so i remove the assert logic here

Reply:
Perfect.